### PR TITLE
Use a custom matcher to assert against CompletionStages.

### PR DESCRIPTION
**Pull request by mbruggmann**

### CODE REVIEW

File: src/test/java/com/spotify/futures/FunctionsTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+package com.spotify.futures;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class FunctionsTest {
```
Comment (dflemstr) [inline src/test/java/com/spotify/futures/FunctionsTest.java]: Unrelated change?

Comment (mbruggmann) [inline src/test/java/com/spotify/futures/FunctionsTest.java]: Kind of. It's two commits at least, but I can split the PR as well if you prefer that.

File: src/test/java/com/spotify/futures/CompletableFuturesTest.java [hunk 2]
```diff
@@ -429,4 +431,23 @@ public void combine5_exceptional() throws Exception {
     exception.expectCause(isA(IllegalStateException.class));
     getCompleted(future);
   }
+
+  private static <T> Matcher<CompletionStage<T>> completesTo(final T expected) {
+    return completesTo(is(expected));
+  }
+
+  private static <T> Matcher<CompletionStage<T>> completesTo(final Matcher<T> expected) {
+    return new CustomTypeSafeMatcher<CompletionStage<T>>("complete to" + String.valueOf(expected)) {
```
Comment (dflemstr) [inline src/test/java/com/spotify/futures/CompletableFuturesTest.java]: Space after `to`

File: src/test/java/com/spotify/futures/CompletableFuturesTest.java [hunk 3]
```diff
@@ -429,4 +431,23 @@ public void combine5_exceptional() throws Exception {
...
+      @Override
+      protected boolean matchesSafely(CompletionStage<T> item) {
+        try {
+          final T value = item.toCompletableFuture().get(1, TimeUnit.SECONDS);
```
Comment (dflemstr) [inline src/test/java/com/spotify/futures/CompletableFuturesTest.java]: You could also depend on `com.jayway.awaitility:awaitility:1.7.0` which provides matchers like this, makes completion time configurable and provides reasons (timeout vs exceptionally completed etc) when it fails.

File: src/test/java/com/spotify/futures/CompletableFuturesTest.java [hunk 4]
```diff
@@ -429,4 +431,23 @@ public void combine5_exceptional() throws Exception {
...
+          return expected.matches(value);
+        } catch (Exception ex) {
```
Comment (dflemstr) [inline src/test/java/com/spotify/futures/CompletableFuturesTest.java]: Would be nice if `ex` somehow got included in the `Description` so that you could diagnose the error without a debugger.

Comment (mbruggmann) [inline src/test/java/com/spotify/futures/CompletableFuturesTest.java]: The current failure in case of a timeout reads like this:
```
java.lang.AssertionError: 
Expected: completes to is "hello"
     but: was <java.util.concurrent.CompletableFuture@4361bd48[Not completed]>
```
I considered that good enough for now, but I'm open for ideas (separate PR?)

**Comment by dflemstr** [comment]
:+1:
